llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -3237,9 +3238,10 @@ bool X86TargetLowering::shouldFormOverflowOp(unsigned Opcode, EVT VT,
 }
 
 bool X86TargetLowering::isCheapToSpeculateCttz(Type *Ty) const {
-  // Speculate cttz only if we can directly use TZCNT or can promote to i32.
+  // Speculate cttz only if we can directly use TZCNT or can promote to i32/i64.
   return Subtarget.hasBMI() ||
-         (!Ty->isVectorTy() && Ty->getScalarSizeInBits() < 32);
+         (!Ty->isVectorTy() &&
+          Ty->getScalarSizeInBits() < (Subtarget.is64Bit() ? 64u : 32u));
```

Comment:
Is there any different to add `u`?

Reply:
It fixes an annoying MSVC warning (I have /WX enabled) - it can't correctly see through the ternary to check the bounds of the int values. I can pull out the max bounds value instead if you prefer?